### PR TITLE
Fix getting env variables

### PR DESCRIPTION
**Pull request by yankouskia**
https://github.com/petkaantonov/bluebird/issues/1294

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -311,18 +311,22 @@ if (typeof Symbol !== "undefined" && Symbol.iterator) {
     asArray = function(v) {
         if (es5.isArray(v)) {
             return v;
-        } else if (v != null && typeof v[Symbol.iterator] === "function") {
+        }
+        if (v != null && typeof v[Symbol.iterator] === "function") {
```
Comment (benjamingr) [inline src/util.js]: Why this change?

File: src/util.js [hunk 2]
```diff
@@ -311,18 +311,22 @@ if (typeof Symbol !== "undefined" && Symbol.iterator) {
...
             return ArrayFrom(v);
         }
         return null;
     };
 }
 
 var isNode = typeof process !== "undefined" &&
-        classString(process).toLowerCase() === "[object process]";
+    classString(process).toLowerCase() === "[object process]";
```
Comment (benjamingr) [inline src/util.js]: why this change?

File: src/util.js [hunk 3]
```diff
@@ -311,18 +311,22 @@ if (typeof Symbol !== "undefined" && Symbol.iterator) {
...
+
+var hasEnvVariables = typeof process !== "undefined" &&
+    typeof process.env !== "undefined";
 
-function env(key, def) {
-    return isNode ? process.env[key] : def;
+function env(key) {
+    return hasEnvVariables ? process.env[key] : undefined;
```
Comment (benjamingr) [inline src/util.js]: change LGTM.

File: src/util.js (hunk 1 above)
Comment (yankouskia) [inline src/util.js]: else is not needed after `return` statement

File: src/util.js (hunk 2 above)
Comment (yankouskia) [inline src/util.js]: tab indentation was broken, but I can return it back if any concerns about it :)

**Comment by deser** [comment]
Great job! Thanks

File: src/util.js (hunk 1 above)
Comment (benjamingr) [inline src/util.js]: Please revert the stylistic changes - I'm +1 on the env variable change.

**Comment by benjamingr** [comment]
@yankouskia can you please squish? 
Thanks

**Comment by yankouskia** [comment]
@benjamingr , ready!

**Comment by yankouskia** [comment]
@benjamingr, thanks!
